plugin-bazel/src/main/kotlin/org/jetbrains/bazel/fastbuild/BazelFastBuildProjectTaskRunner.kt

```diff
@@ -0,0 +1,70 @@
+package org.jetbrains.bazel.fastbuild
+
+import com.intellij.ide.impl.isTrusted
+import com.intellij.openapi.application.runInEdt
+import com.intellij.openapi.components.service
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.registry.Registry
+import com.intellij.task.ModuleBuildTask
+import com.intellij.task.ModuleFilesBuildTask
+import com.intellij.task.ProjectTask
+import com.intellij.task.ProjectTaskContext
+import com.intellij.task.ProjectTaskRunner
+import org.jetbrains.bazel.config.isBazelProject
+import org.jetbrains.bazel.coroutines.BazelCoroutineService
+import org.jetbrains.bazel.server.connection.connection
+import org.jetbrains.bazel.settings.bazel.bazelProjectSettings
+import org.jetbrains.bazel.target.TargetUtils
+import org.jetbrains.bsp.protocol.FastBuildParams
+import org.jetbrains.concurrency.AsyncPromise
+import org.jetbrains.concurrency.Promise
+import org.jetbrains.concurrency.all
+import java.nio.file.Files
+
+class BazelFastBuildProjectTaskRunner: ProjectTaskRunner() {
+  override fun canRun(projectTask: ProjectTask): Boolean {
+    return when (projectTask) {
+      is ModuleFilesBuildTask -> !projectTask.module.project.bazelProjectSettings.enableBuildWithJps
+      else -> false
+    }
+  }
+
+  override fun canRun(
+    project: Project,
+    projectTask: ProjectTask,
+    context: ProjectTaskContext?
+  ): Boolean {
+    return Registry.`is`(FastBuildUtils.fastBuildEnabledKey) &&
+      project.isBazelProject &&
+      project.isTrusted() &&
+      canRun(projectTask)
+  }
+
+  override fun run(
+    project: Project,
+    context: ProjectTaskContext,
+    vararg tasks: ProjectTask?
+  ): Promise<Result> {
+    val moduleBuildTasks = tasks.filterIsInstance<ModuleFilesBuildTask>()
+    val result = AsyncPromise<Result>()
+    BazelCoroutineService.getInstance(project).startAsync {
+      runInEdt {
+        FileDocumentManager.getInstance().saveAllDocuments()
+      }
+      val targetUtils = project.service<TargetUtils>()
+      val results = ArrayList<Promise<Result>>()
+      moduleBuildTasks.flatMap { it.files.toList() }.forEach { file ->
+        val tempDir = Files.createTempDirectory(file.name)
```

Comment:
This temp directory will only be deleted once user clears the `Temp` dir on their machine, should we actually delete it sometime?